src/java/org/apache/cassandra/index/sai/disk/v1/SSTableIndexWriter.java

```diff
@@ -319,20 +321,52 @@ private void writeSegmentsMetadata() throws IOException
         }
     }
 
-    private SegmentBuilder newSegmentBuilder(long rowIdOffset)
+    private SegmentBuilder newSegmentBuilder(long rowIdOffset) throws IOException
     {
         SegmentBuilder builder;
 
         if (indexContext.isVector())
         {
             // if we have a PQ instance available, we can use it to build a CompactionGraph;
             // otherwise, build on heap (which will create PQ for next time, if we have enough vectors)
-            var pqi = CassandraOnHeapGraph.getPqIfPresent(indexContext, vc -> vc.type == VectorCompression.CompressionType.PRODUCT_QUANTIZATION);
-            if (pqi == null || pqi.unitVectors.isEmpty() || !V3OnDiskFormat.ENABLE_LTM_CONSTRUCTION) {
+            var pqi = CassandraOnHeapGraph.getPqIfPresent(indexContext, vc -> vc.type == CompressionType.PRODUCT_QUANTIZATION);
+            if (pqi == null && segments.size() > 0)
+            {
+                // No PQ instance available in completed indexes, so check if we just wrote one
+                var pqComponent = perIndexComponents.get(IndexComponentType.PQ);
+                assert pqComponent != null; // we always have a PQ component even if it's not actually PQ compression
+                try (var fh = pqComponent.createFileHandle();
+                     var reader = fh.createReader())
+                {
+                    var sm = segments.get(segments.size() - 1);
```

Comment:
Just to make sure I understand this, we use the last segment's PQ so that if we are building an index with 3 segments, the third segment's PQ will benefit from the first two segments, right?

Reply:
When we're on the LTM path (CompactionGraph instead of CassandraOnHeapGraph) we use the PQ as-is.  We should still converge on better PQ as new data is written through COHG but it won't be as fast.

That said, VectorSiftSmallTest exercises this path and checks for recall degradation, it seems okay in practice as well as in theory.

Reply:
Yes, it's equivalent to get(0) for now.

But if we figure out how to update PQ in CompactionGraph, there's one less thing to remember to update.